src/systems/physics/Physics.cc

```diff
@@ -1117,6 +1157,37 @@ void PhysicsPrivate::CreatePhysicsEntities(const EntityComponentManager &_ecm)
         }
         return true;
       });
+
+  _ecm.EachNew<components::EnableContactSurfaceCustomization,
+               components::Collision, components::Name>(
+      [&](const Entity & _entity,
+          const components::EnableContactSurfaceCustomization *_enable,
+          const components::Collision */*_collision*/,
+          const components::Name *_name) -> bool
+      {
+        const auto world = worldEntity(_entity, _ecm);
+        if (_enable)
+        {
+          if (this->customContactSurfaceEntities[world].empty())
+          {
+            this->EnableContactSurfaceCustomization(world);
+          }
+          this->customContactSurfaceEntities[world].insert(_entity);
+          ignmsg << "Enabling contact surface customization for collision ["
+                 << _name->Data() << "]" << std::endl;
+        }
+        else
+        {
+          this->customContactSurfaceEntities[world].erase(_entity);
+          ignmsg << "Disabling contact surface customization for collision ["
+                 << _name->Data() << "]" << std::endl;
+          if (this->customContactSurfaceEntities[world].empty())
+          {
+            this->DisableContactSurfaceCustomization(world);
```

Comment:
Can you add a test that covers this? As it's currently implemented, I don't think this is working. We're checking `_enable` , which is a pointer to the component, not the data. `Each*` functions never call the callback with a `nullptr` component.

Reply:
Good catch. I also needed to change `EachNew` to `Each` because I misunderstood that `EachNew` only triggers for newly added entities, and not just for newly added components to existing entities. But `Each` is okay here because the component is removed after being processed.

I added the test (removing the components and checking that the vehicle did not move).